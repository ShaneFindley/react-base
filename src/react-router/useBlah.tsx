export function useBlah() {
  return {
    data: [] as string[],
    error: undefined
  }
}

export function Comp() {
  const {
    data,
    error
  } = useBlah();
  return <div>
        {data.map(d => <div>{d}</div>)}
        {error && <div>{error}</div>}
    </div>;
}
