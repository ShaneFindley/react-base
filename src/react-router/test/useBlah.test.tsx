import { render } from "@testing-library/react";
import { Comp, useBlah } from "../useBlah";

jest.mock("../useBlah", () => ({
  ...jest.requireActual("../useBlah"),
  useBlah: jest.fn(() => ({ data: [] }))
}))

beforeEach(() => {
  jest.clearAllMocks();
})

it('shekjf', () => {
  const { container } = render(<Comp />);
  expect(container).toMatchSnapshot();
})

it('awdjgyawd', () => {
  jest.mocked(useBlah).mockReturnValue({ data: ['test1', 'test2'], error: undefined })
  const { container } = render(<Comp />);
  expect(container).toMatchSnapshot();
})
