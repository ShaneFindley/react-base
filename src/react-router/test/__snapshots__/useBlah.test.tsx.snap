// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`awdjgyawd 1`] = `
<div>
  <div />
</div>
`;

exports[`shekjf 1`] = `
<div>
  <div />
</div>
`;
